nav2_costmap_2d/CMakeLists.txt

```diff
@@ -141,34 +147,30 @@ ament_target_dependencies(nav2_costmap_2d
 )
 
 target_link_libraries(nav2_costmap_2d
-  nav2_costmap_2d_core
+  ${PROJECT_NAME}::nav2_costmap_2d_core
   layers
-  filters
+  ${PROJECT_NAME}::filters
 )
 
 install(TARGETS
-  nav2_costmap_2d_core
   layers
   filters
+  nav2_costmap_2d_core
   nav2_costmap_2d_client
+  nav2_costmap_2d_markers
+  nav2_costmap_2d_cloud
+  EXPORT export_${PROJECT_NAME}
   ARCHIVE DESTINATION lib
   LIBRARY DESTINATION lib
   RUNTIME DESTINATION bin
 )
 
-install(TARGETS
```

Comment:
Readd. The install locations of executables and libraries are different.

Reply:
Re-added. You can do different destinations for both in the same `install` command by changing the `RUNTIME DESTINATION`, however it looks like this causes the nodes to be exported in the export set which is would break consumers using ament_target_dependencies because you can't link to an executable.

Existing behavior is preserved for two installs, which is in line with the tutorials.